CromIAM/src/main/scala/cromiam/webservice/SubmissionSupport.scala

```diff
@@ -30,37 +33,52 @@ trait SubmissionSupport extends RequestSupport {
-              forwardSubmissionToCromwell(user, submission.collection, request.withEntity(submission.entity))
+        onComplete(samClient.isSubmitWhitelisted(user, request).value.unsafeToFuture()) {
+          case Success(Left(httpResponse)) => complete(httpResponse)
+          case Success(Right(submitWhitelisted)) =>
+            authorize(submitWhitelisted) {
+              extractSubmission(user) { submission =>
+                complete {
+                  forwardSubmissionToCromwell(
+                    user,
+                    submission.collection,
+                    request.withEntity(submission.entity)
+                  ).asHttpResponse
+                }
+              }
             }
-          }
+          case Failure(e) =>
+            val message = s"Unable to look up submit whitelist for user ${user.userId}: ${e.getMessage}"
+            throw new RuntimeException(message, e)
         }
       }
     }
   }
 
   private def forwardSubmissionToCromwell(user: User,
                                           collection: Collection,
-                                          submissionRequest: HttpRequest): Future[HttpResponse] = {
+                                          submissionRequest: HttpRequest): FailureResponseOrT[HttpResponse] = {
     log.info("Forwarding submission request for " + user.userId + " with collection " + collection.name + " to Cromwell")
 
-    def registerWithSam(collection: Collection, httpRequest: HttpRequest): Future[Unit] = {
-      samClient.requestSubmission(user, collection, httpRequest) recoverWith {
-        case e: SamDenialException => Future.failed(e)
-        case SamRegisterCollectionException(statusCode) => Future.failed(SamRegisterCollectionException(statusCode))
-        case e => Future.failed(SamConnectionFailure("new workflow registration", e))
-      }
+    def registerWithSam(collection: Collection, httpRequest: HttpRequest): FailureResponseOrT[Unit] = {
+      FailureResponseOrT(
+        samClient.requestSubmission(user, collection, httpRequest).value handleErrorWith {
+          case e: SamDenialException => IO.raiseError(e)
+          case SamRegisterCollectionException(statusCode) => IO.raiseError(SamRegisterCollectionException(statusCode))
+          case e => IO.raiseError(SamConnectionFailure("new workflow registration", e))
+        }
+      )
     }
 
-    (for {
-      _ <- registerWithSam(collection, submissionRequest)
-      resp <- cromwellClient.forwardToCromwell(submissionRequest)
-    } yield resp) recover {
-      case _: SamDenialException => SamDenialResponse
-      case SamRegisterCollectionException(statusCode) => SamRegisterCollectionExceptionResp(statusCode)
-    }
+    FailureResponseOrT(
+      (for {
+        _ <- registerWithSam(collection, submissionRequest)
+        resp <- cromwellClient.forwardToCromwell(submissionRequest)
+      } yield resp).value handleErrorWith {
+        case _: SamDenialException => IO.pure(Left(SamDenialResponse))
```

Comment:
ToL note to self to come back here and log these Sam Denial Exceptions.  Acknowledging they are not in scope for this PR